### PR TITLE
Tests and basic operation

### PR DESCRIPTION
**Pull request by AshishS-1123**
# Summary

- Created basic feature for Reading and writing images.
- Created setup for tests.

Developed an abstract class to represent an operation. Right now, this class contains very little code, which will be improved in later PRs.

Unittests have also been developed. To run tests run, ```ninja test``` from the **build** folder.
I have also made some changes to the meson file, added **gdk** and **gdkpixbuf** as dependencies. 

Future work will involve laying down some groundwork for developing brushes and image operations. This will generally involve major changes to the **AbstractOperation** class.

### CODE REVIEW

File: src/ImageIO/Writer.vala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LibPhotostat. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Ashish Shevale <shevaleashish@gmail.com>
+ */
+
+public LibPhotostat.ImageIO.Writer {
+    public Writer () {
```
Comment (ghost) [inline src/ImageIO/Writer.vala]: Add `class` after public

File: src/ImageIO/Displayer.vala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LibPhotostat. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Ashish Shevale <shevaleashish@gmail.com>
+ */
+
+public LibPhotostat.ImageIO.Displayer {
```
Comment (ghost) [inline src/ImageIO/Displayer.vala]: add `class` here too

**Comment by AshishS-1123** [comment]
@Abdallah-Moh can you take a look at the meson file. I had to make some changes and removed a lot of things because I was getting a compile error. Make sure ```ninja test``` works afterwards.

**Comment by ghost** [comment]
It works

**Review by colinkiama — CHANGES_REQUESTED** [review]
AbstractOperation should marked as an **abstract** class.

For reference, see the "Abstract Classes" section in the Vala tutorial: https://wiki.gnome.org/Projects/Vala/Tutorial#Abstract_Classes

**Review by colinkiama — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**